core/src/main/java/uk/ac/bbsrc/tgac/miso/core/data/Experiment.java

```diff
@@ -12,231 +12,369 @@
-   * @return Platform platform.
-   */
-  public Platform getPlatform();
+  public String getDescription() {
+    return description;
+  }
 
-  /**
-   * Sets the platform of this Experiment object.
-   * 
-   * @param platform
-   *          platform.
-   * 
-   */
-  public void setPlatform(Platform platform);
+  @Override
+  public long getId() {
+    return experimentId;
+  }
+
+  public Collection<Kit> getKits() {
+    return kits;
+  }
+
+  public Collection<Kit> getKitsByKitType(KitType kitType) {
+    final ArrayList<Kit> ks = new ArrayList<>();
+    for (final Kit k : kits) {
+      if (k.getKitDescriptor().getKitType().equals(kitType)) {
+        ks.add(k);
+      }
+    }
+    Collections.sort(ks);
+    return ks;
+  }
+
+  public User getLastModifier() {
+    return lastModifier;
+  }
+
+  public Library getLibrary() {
+    return library;
+  }
 
-  /**
-   * Returns the run of this Experiment object.
-   * 
-   * @return Run run.
-   */
-  public Run getRun();
+  @Override
+  public String getName() {
+    return name;
+  }
 
-  /**
-   * Sets the run of this Experiment object.
-   * 
-   * @param run
-   *          run.
-   * 
-   */
-  public void setRun(Run run);
+  public Platform getPlatform() {
+    return platform;
+  }
 
-  /**
-   * Sets the pool of this Experiment object.
-   * 
-   * @param pool
-   *          pool.
-   * 
-   */
-  public void setPool(Pool pool);
+  public List<RunPartition> getRunPartitions() {
+    return runPartitions;
+  }
 
-  /**
-   * Returns the pool of this Experiment object.
-   * 
-   * @return Pool pool.
-   */
-  public Pool getPool();
+  @Override
+  public SecurityProfile getSecurityProfile() {
+    return securityProfile;
+  }
 
-  /**
-   * Sets the kits of this Experiment object.
-   * 
-   * @param kits
-   *          kits.
-   */
-  public void setKits(Collection<Kit> kits);
+  public Study getStudy() {
+    return study;
+  }
 
-  /**
-   * Add a Kit to this Experiment object
-   * 
-   * @param kit
-   *          of type Kit
-   */
-  public void addKit(Kit kit);
+  public String getTitle() {
+    return title;
+  }
 
-  /**
-   * Returns the kits used to construct this Experiment object.
-   * 
-   * @return Collection<Kit> kits.
-   */
-  public Collection<Kit> getKits();
+  @Override
+  @CoverageIgnore
+  public int hashCode() {
+    if (getId() != UNSAVED_ID) {
+      return (int) getId();
+    } else {
+      final int PRIME = 37;
+      int hashcode = 1;
+      if (getName() != null) hashcode = PRIME * hashcode + getName().hashCode();
+      if (getAlias() != null) hashcode = 37 * hashcode + getAlias().hashCode();
+      return hashcode;
+    }
+  }
 
-  /**
-   * Return kits used within this experiment of a given KitType
-   * 
-   * @param kitType
-   *          of type KitType
-   * @return Collection<Kit>
-   */
-  public Collection<Kit> getKitsByKitType(KitType kitType);
+  @Override
+  public void inheritPermissions(SecurableByProfile parent) throws SecurityException {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove the declaration of thrown exception 'java.lang.SecurityException' which is a runtime exception. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ARedundantThrowsDeclarationCheck)